core/src/main/java/org/jdbi/v3/core/EnumConfig.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core;
+
+import org.jdbi.v3.core.config.JdbiConfig;
+
+/**
+ * Configuration for behavior related to {@link Enum}s.
+ */
+public class EnumConfig implements JdbiConfig<EnumConfig> {
+    private boolean handleEnumsByName = true;
```

Comment:
Would it be useful to have this be a `private Annotation defaultEnumQualifier`, a more general mechanism that doesn't explicitly encode a choice between the two built in strategies?

Reply:
Are you talking about the API or internal workings? We can change the boolean internals to something tertiary when the need arises. The binary nature of the API did bother me too though. I was thinking an enum would be a less surprising thing for users to have to use than `setEnumHandling(EnumByName.class)`

Reply:
Actually, I take that back. `setEnumHandling(EnumByName.class)` is probably pretty neat, less redundant stuff for users to have to link together since they'll just be using the annotations as points of contact for everything to do with the feature.

Reply:
I think I would like a marker interface on those annotations then though, so you can't pass just any `Annotation` to the methods. That'd be like going back to java 5.

Reply:
We can't make that method value-safe though, users would be able to pass in just any annotation class. I'd go with an enum that collects the different annotations as values, only to be used for safety on the setter.